src/main/java/net/glowstone/entity/EntityRegistry.java

```diff
@@ -159,107 +166,172 @@
+        put(Pig.class, GlowPig.class, GlowPig::new);
+        put(PigZombie.class, GlowPigZombie.class, GlowPigZombie::new);
+        put(Player.class, GlowPlayer.class, location -> {
+            throw new UnsupportedOperationException(
+                    "Attempt to spawn a Player with no associated login");
+        });
+        put(PolarBear.class, GlowPolarBear.class, GlowPolarBear::new);
+        put(TNTPrimed.class, GlowTntPrimed.class, GlowTntPrimed::new);
+        put(Rabbit.class, GlowRabbit.class, GlowRabbit::new);
+        put(Sheep.class, GlowSheep.class, GlowSheep::new);
+        put(Shulker.class, GlowShulker.class, GlowShulker::new);
+        put(Silverfish.class, GlowSilverfish.class, GlowSilverfish::new);
+        put(Skeleton.class, GlowSkeleton.class, GlowSkeleton::new);
+        put(SkeletonHorse.class, GlowSkeletonHorse.class, GlowSkeletonHorse::new);
+        put(Slime.class, GlowSlime.class, GlowSlime::new);
+        put(Snowball.class, GlowSnowball.class, GlowSnowball::new);
+        put(Snowman.class, GlowSnowman.class, GlowSnowman::new);
+        put(SpectralArrow.class, GlowSpectralArrow.class, GlowSpectralArrow::new);
+        put(Spider.class, GlowSpider.class, GlowSpider::new);
+        put(SplashPotion.class, GlowSplashPotion.class, GlowSplashPotion::new);
+        put(Squid.class, GlowSquid.class, GlowSquid::new);
+        put(Stray.class, GlowStray.class, GlowStray::new);
+        put(ThrownExpBottle.class, GlowThrownExpBottle.class, GlowThrownExpBottle::new);
+        put(TippedArrow.class, GlowTippedArrow.class, GlowTippedArrow::new);
+        put(Vex.class, GlowVex.class, GlowVex::new);
+        put(Villager.class, GlowVillager.class, GlowVillager::new);
+        put(Vindicator.class, GlowVindicator.class, GlowVindicator::new);
+        put(Weather.class, GlowWeather.class, GlowLightningStrike::new);
+        put(Witch.class, GlowWitch.class, GlowWitch::new);
+        put(Wither.class, GlowWither.class, GlowWither::new);
+        put(WitherSkeleton.class, GlowWitherSkeleton.class, GlowWitherSkeleton::new);
+        //TODO: Wither Skull
+        put(Wolf.class, GlowWolf.class, GlowWolf::new);
+        put(Zombie.class, GlowZombie.class, GlowZombie::new);
+        put(ZombieHorse.class, GlowZombieHorse.class, GlowZombieHorse::new);
+        put(ZombieVillager.class, GlowZombieVillager.class, GlowZombieVillager::new);
+        ENTITIES = entitiesBuilder.build();
+        ENTITY_CTORS = entityCtorsBuilder.build();
+    }
+
+    /**
+     * Creates an entity of the given type at the given location.
+     *
+     * @param type     the entity type
+     * @param location the location to spawn the entity
+     * @return an entity that has not yet been sent to clients
+     */
+    public static GlowEntity constructEntity(Class<? extends Entity> type, Location location) {
+        if (ENTITY_CTORS.containsKey(type)) {
+            return ENTITY_CTORS.get(type).apply(location);
```

Comment:
Actually, I just realized from the below line that EntityStore is essentially a factory for entities, with its createEntity() method. We should just use that rather than duplicating that here, then custom entities will be handled the same as builtins.

Reply:
Refactored; see https://github.com/Pr0methean/Glowstone/commit/6baa9be7f050dfb50a037da51f5f7613cb8083e7#diff-8769715641781a189890669ce58619a0R173